keras_cv/layers/preprocessing/with_bounding_boxes_test.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import tensorflow as tf
+from absl.testing import parameterized
+
+from keras_cv.layers import preprocessing
+
+# Imports from with_labels the list of augmentation that should perform a No-Op.
+from keras_cv.layers.preprocessing.with_labels_test import TEST_CONFIGURATIONS
```

Comment:
we aren't guaranteed to pass all with_bounding_box_tests based on with_labels test, please duplicate the list here.

Reply:
> Maybe we have a test file meant for storing a last of these and import from there, with a list of geometric layers and a list of non-geometric layers?

I moved all the generic configuration in one file `layers_configurations_test.py`

As stated in the file comment the `TEST_CONFIGURATIONS` variable should only contains layers that perform No-Op operations on bounding boxes and keypoints. the `GEOMETRIC_TEST_CONFIGURATIONS` will list layers that apply transformation to boxes / keypoints, but without the needed parameter. 

When a new preprocessing layer is created, it should be added in one of these two list, depending on the fact that augment_bounding_boxes is a No-Op or no.

`with_labels.py` test for label supports on both list
`with_bounding_boxes.py` and `with_keypoints.py` only imports the `TEST_CONFIGURATIONS` list to check for no-op. They redefine their own `GEOMETRIC_TEST_CONFIGURATIONS` list to test for bounding box support.

The rationales behind this organization are:
* One can add a new layer that has no bounding box/keypoint support yet, but would be added in the future and therefore should this support is not tested yet. Please note that this is required to make tests pass while breaking down keypoint support for each layer independantly
* We can test that layer that performs geometric manipulation works on labels even if no `bounding_box_format` / `keypoint_format` argument are passed, as it is a feature of the API. 

Bump @LukeWood  for another review.